lib/widgets/actions.dart

```diff
@@ -21,41 +21,63 @@ Future<bool> markNarrowAsRead(
   Narrow narrow,
   bool useLegacy, // TODO(server-6)
 ) async {
-  try {
-    final store = PerAccountStoreWidget.of(context);
-    final connection = store.connection;
-    if (useLegacy) {
-      await _legacyMarkNarrowAsRead(context, narrow);
-      return true;
-    }
+  if (useLegacy) {
+    await _legacyMarkNarrowAsRead(context, narrow);
```

Comment:
Now that the the try-catch has been moved inside the helper, making the call here can still throw. This is no longer an nfc. Let's check if there's some more handling needed here.